### PR TITLE
refactor: Remove most of `body_part` enum usage

### PR DESCRIPTION
**Pull request by Coolthulhu**
### Required

- [x] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [x] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [x] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change

Working towards removable limbs

## Describe the solution

Grunt work consisting of:
* Replacing iteration over the enum to iteration over player's body parts
* Replacing `num_bp` comparisons with null id checks

## Testing

Things that need to be specifically tested:
* Wearing power armor accessories respecting limits on 2 accessory per bp
* Switching sides on sided equipment respecting usual limits
* Body part hp display
* Body part encumbrance description in `@` menu
* Layering penalty
* NPCs being given splints
* Monster basic melee attack printing correct messages regarding player's body parts
* Shotgun trap picking different body parts
* Triffid queen attack damaging different body parts

## Additional context

* [x] Rebase after #5685 is merged

### CODE REVIEW

**Review by chaosvolt — COMMENTED** [review]
1. Compiled and load-tested, no oddities seen with an old save.
2. Spawned some lava nearby, checked that heat gain/loss and blistering seemed to work right.
3. Put on the white glove, confirmed right hand specifically was warmer than left hand.
4. Put on breeches, a tunic, a duster, chaps, and knee-high boots. Compact flag still worked.
5. Put on two pairs of bone greaves, got the encumbrance penalty as expected.
6. Confirmed I can't put on three pairs of bone greaves.
7. Confirmed I can put on power armor and wear a helmet over it.

One regression: power armor accessories no longer limit what you can put on each bodypart.

image

I'm mixed on this however. Prior behavior was kinda poorly-done, it treated "either leg/arm" as a single bodypart and thus forbid you from wearing a power armor leg sheath on one leg and a leg mount on the other, when sane behavior would be for the "PA accessories are limited to one per body part" to correctly interpret "either side" as only one part, not both at once even after it's been assigned a side.

**Comment by chaosvolt** [comment]
Oh wait, I tested in my old desktop build `CBN 2024-11-26` and it's doing this too, dafuq.
image

I thought power armor accessories were supposed to be limited to one per limb, hence the earlier discovery that it was bugged? And now that bug has gone and vanished since last time I encountered it... XD

**Comment by Coolthulhu** [comment]
> "PA accessories are limited to one per body part"

The code suggests it was supposed to be 2 accessories per bp.

**Comment by chaosvolt** [comment]
Then guess it's working as intended yeah, will resume testing in a bit when I get home.

**Review by chaosvolt — APPROVED** [review]
Continued testing:
1. Encumberance display on the `@` screen looks right, also slapped on heavy survivor set to see what that does and seems to do the right stuff.
2. Spawned a zombie predator, it flailed at me uselessly (heavy survivor gear woot) until biting the leg leg which correctly showed as injured.
3. Waited until it impaled me, it said it hit the torso and that's what's showing as injured and bleeding.
4. Bandaged my torso, bleeding was stopped and the bandaged effect is on the correct bodypart.
5. Inflicted myself with eepy and took a nap, torso healed up due to being bandaged as expected.
6. Recruited NPC friend, broke one of their legs and made them put on a splint, they wore it on the correct leg as expected.
7. Waited several days, examined the NPC to find the limb was making progress on recovering.
8. Blindfolded myself, spawned a line of shotgun traps and walked through them, different parts get hit and message matches the part taking damage.
9. Stripped naked and spawned in a triffid queen, it would impale different parts when I get dinged by a tree spawning and the expected part takes damage.
10. Returned to my heavy survivor set, put it on, and spawned a second pair of heavy survivor gloves to wear. My hands correctly take a bonus encumbrance penalty.

**Comment by scarf005** [comment]
is it good to merge?

**Comment by chaosvolt** [comment]
Unless anyone can think of anything else to test, it seems good to go.

**Comment by scarf005** [comment]
lets goooooo

File: src/armor_layers.cpp [hunk 1]
```diff
@@ -542,7 +542,6 @@ void show_armor_layers_ui( Character &who )
     ctxt.register_action( "NEXT_TAB" );
     ctxt.register_action( "MOVE_ARMOR" );
     ctxt.register_action( "CHANGE_SIDE" );
-    ctxt.register_action( "TOGGLE_CLOTH" );
```
Comment (scarf005) [inline src/armor_layers.cpp:545]: oops

**Pull request merged**